src/generic.jl

```diff
@@ -74,3 +74,16 @@ function pairwise(d::ImageSemiMetric, imgs::AbstractArray{IMG}) where {IMG<:Abst
 
     D
 end
+
```

Comment:
I need to use `difftype` defined in `Images` to ensure the correct numerical result, should I move it here or move it to `ImageCore`? I prefer the latter because it can be used for all other similar calculations.

https://github.com/JuliaImages/Images.jl/issues/765